spanner/spanner_snippets/spanner/pg_spanner_numeric_data_type.go

```diff
@@ -0,0 +1,162 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spanner
+
+// [START spanner_postgresql_numeric_data_type]
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"cloud.google.com/go/spanner"
+	database "cloud.google.com/go/spanner/admin/database/apiv1"
+	"google.golang.org/api/iterator"
+	adminpb "google.golang.org/genproto/googleapis/spanner/admin/database/v1"
+)
+
+// pgNumericDataType shows how to work with the PostgreSQL NUMERIC/DECIMAL data
+// type on a Spanner PostgreSQL database.
+func pgNumericDataType(w io.Writer, db string) error {
+	// db := "projects/my-project/instances/my-instance/databases/my-database"
+	ctx := context.Background()
+
+	adminClient, err := database.NewDatabaseAdminClient(ctx)
+	if err != nil {
+		return err
+	}
+	defer adminClient.Close()
+
+	// Create a table that includes a column with data type NUMERIC. As the database has been
+	// created with the PostgreSQL dialect, the data type that is used will be the PostgreSQL
+	// NUMERIC / DECIMAL data type.
+	op, err := adminClient.UpdateDatabaseDdl(ctx, &adminpb.UpdateDatabaseDdlRequest{
+		Database: db,
+		Statements: []string{
+			`CREATE TABLE Venues (
+				VenueId  bigint NOT NULL PRIMARY KEY,
+				Name     varchar(1024) NOT NULL,
+				Revenues numeric
+			)`},
+	})
+	if err != nil {
+		return err
+	}
+	if err := op.Wait(ctx); err != nil {
+		return err
+	}
+	fmt.Fprintf(w, "Created Venues table\n")
+
+	client, err := spanner.NewClient(ctx, db)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	var updateCount int64
+	// Insert a Venue using DML.
+	_, err = client.ReadWriteTransaction(ctx, func(ctx context.Context, transaction *spanner.ReadWriteTransaction) error {
+		updateCount, err = transaction.Update(ctx, spanner.Statement{
+			SQL: `INSERT INTO Venues (VenueId, Name, Revenues) VALUES ($1, $2, $3)`,
+			Params: map[string]interface{}{
+				"p1": 1,
+				"p2": "Venue 1",
+				"p3": spanner.PGNumeric{Numeric: "3150.25", Valid: true},
+			},
+		})
+		return err
+	})
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(w, "Inserted %d venue(s)\n", updateCount)
+
+	// Insert a Venue with a NULL value for the Revenues column.
+	_, err = client.ReadWriteTransaction(ctx, func(ctx context.Context, transaction *spanner.ReadWriteTransaction) error {
+		updateCount, err = transaction.Update(ctx, spanner.Statement{
+			SQL: `INSERT INTO Venues (VenueId, Name, Revenues) VALUES ($1, $2, $3)`,
+			Params: map[string]interface{}{
+				"p1": 2,
+				"p2": "Venue 2",
+				"p3": spanner.PGNumeric{Valid: false},
+			},
+		})
+		return err
+	})
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(w, "Inserted %d venue(s) with NULL revenues\n", updateCount)
+
+	// Insert a Venue with a NaN (Not a Number) value for the Revenues column.
```

Comment:
This sample shows a lot of stuff! It looks like it demonstrates several things:

+ How to insert an actual number
+ How to insert a NULL value
+ How to insert a NaN value

I think that this will be hard to parse when published to cloud.google.com. I recommend providing multiple region tags so that each example (numeric, NULL, NaN) can be published independently.